compiler/next/include/chpl/AST/PrimCallExpr.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHPL_AST_PRIMCALLEXPR_H
+#define CHPL_AST_PRIMCALLEXPR_H
+
+#include "chpl/AST/CallExpr.h"
+
+#include <vector>
+
+namespace chpl {
+namespace ast {
+
+/**
+  This class represents a primitive call
+ */
+class PrimCallExpr final : public CallExpr {
+ private:
+  bool matchesInner(const BaseAST* other) const override;
+ public:
+  ~PrimCallExpr() override = default;
+
+  // TODO: which primitive?
+  int numActuals() const override { return this->numChildren(); }
+  Expr* actual(int i) const override { return this->getChild(i); }
+};
```

Comment:
I don't want any of `Call` subclasses to override these. The motivation being to avoid performance impact (dynamic dispatch) and to make it easier to factor out all support for handling the actual arguments into `Call`.

For `PrimCall` and `OpCall`, have child(0) either be NULL or point to the Expr for the primitive or the operator.

Reply:
Sure. I've already removed these in my branch for a follow-up PR.